### PR TITLE
Implement hard liveliness check

### PR DESCRIPTION
**Pull request by pablogs9**
- [x] Document this feature before merging: https://github.com/eProsima/Micro-XRCE-DDS-docs/pull/72

CI: https://github.com/eProsima/Micro-XRCE-DDS/pull/136

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/548/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/566/)

File: include/uxr/client/config.h.in [hunk 1]
```diff
@@ -67,5 +67,18 @@
 
 #cmakedefine UCLIENT_TWEAK_XRCE_WRITE_LIMIT
 
+#cmakedefine UCLIENT_HARD_LIVELINESS_CHECK
+
+#ifdef UCLIENT_HARD_LIVELINESS_CHECK
+#define UXR_CONFIG_HARD_LIVELINESS_CHECK_TIMEOUT         @UCLIENT_HARD_LIVELINESS_CHECK_TIMEOUT@
```
Comment (Acuadros95) [inline include/uxr/client/config.h.in:73]:
```suggestion
```

File: include/uxr/client/core/type/xrce_types.h [hunk 2]
```diff
@@ -40,7 +41,24 @@ extern "C"
...
+#define HARD_LIVELINESS_CHECK_SEQ_COUNT 1
+#else
+#define HARD_LIVELINESS_CHECK_SEQ_COUNT 0
+#endif // ifdef UCLIENT_HARD_LIVELINESS_CHECK
+
+#if  (PROFILE_SHARED_MEMORY_SEQ_COUNT + HARD_LIVELINESS_CHECK_SEQ_COUNT) == 0
 #define UXR_PROPERTY_SEQUENCE_MAX          1
+#else
+#define UXR_PROPERTY_SEQUENCE_MAX          PROFILE_SHARED_MEMORY_SEQ_COUNT + HARD_LIVELINESS_CHECK_SEQ_COUNT
+#endif // if  (PROFILE_SHARED_MEMORY_SEQ_COUNT + HARD_LIVELINESS_CHECK_SEQ_COUNT) == 0
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply include/uxr/client/core/type/xrce_types.h:61]: I'll keep this way in order to ensure extensibility

File: include/uxr/client/config.h.in (hunk 1 above)
Comment (pablogs9) [reply include/uxr/client/config.h.in]: 964971495713c82b9044489548b48429ef09d05f

File: src/c/core/session/session_info.c [hunk 3]
```diff
@@ -54,12 +56,35 @@ void uxr_buffer_create_session(
...
+#ifdef UCLIENT_HARD_LIVELINESS_CHECK
+    payload.client_representation.optional_properties = true;
+    payload.client_representation.properties.data[payload.client_representation.properties.size].name = "uxr_hl";
+
+    const char* str = UXR_CONFIG_HARD_LIVELINESS_CHECK_TIMEOUT_STR;
+
+    if (UXR_CONFIG_HARD_LIVELINESS_CHECK_TIMEOUT > 999999)
+    {
+        str = "999999";
+    }
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply src/c/core/session/session_info.c]: 964971495713c82b9044489548b48429ef09d05f

File: include/uxr/client/config.h.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (pablogs9) [reply include/uxr/client/config.h.in]: 964971495713c82b9044489548b48429ef09d05f

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/550/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/568/)

**Comment by Acuadros95** [comment]
LGTM

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**